ufl/integral.py

```diff
@@ -30,8 +36,30 @@ class Integral:
         "_ufl_domain",
     )
 
-    def __init__(self, integrand, integral_type, domain, subdomain_id, metadata, subdomain_data):
-        """Initialise."""
+    def __init__(
+        self,
+        integrand: Expr,
+        integral_type: str,
+        domain,
+        subdomain_id: int | tuple[int, ...],
+        metadata: dict,
+        subdomain_data: Any,
+        extra_domain_integral_type_map: dict | None = None,
```

Comment:
Could you be more specific about what the types of the keys and values of this dict is? 
It's quite hard to deduce that from the code.